File: frontend/src/model/configModel.ts
export type AppThemeModel = "light" | "dark";

export type ConfigModel = {
	OllamaEndpoints: OllamaEndpoint[];
	DefaultOllamaEndPointName: string;
};

export type OllamaEndpoint = {
	EndpointName: string;
	EndpointUrl: string;
	LLMModels: string[];
	DefaultLLMModel: string;
};

export type CurrentOllamaHostModel = {
	DisplayName: string;
	Endpoint: string;
	ModelName: string;
};

File: pkg/model/config-model.go
package model

type ConfigJson struct {
	AppTheme                  string           `json:"AppTheme"`
	OllamaEndpoints           []OllamaEndpoint `json:"OllamaEndpoints"`
	DefaultOllamaEndPointName string           `json:"DefaultOllamaEndPointName"`
	DefaultPrompt             string           `json:"DefaultPrompt"`
}

type OllamaEndpoint struct {
	EndpointName    string   `json:"EndpointName"`
	EndpointURL     string   `json:"EndpointUrl"`
	LLMModels       []string `json:"LLMModels"`
	DefaultLLMModel string   `json:"DefaultLLMModel"`
}
